nipype/interfaces/dtitk/registration.py

```diff
@@ -165,7 +165,8 @@ class ComposeXfmInputSpec(CommandLineInputSpec):
                   argstr="-aff %s", mandatory=True)
     out_file = File(desc='output path',
                     argstr="-out %s",  name_source="in_df",
-                    name_template="%s_aff.df", keep_extension=True)
+                    name_template="%s_affdf.nii")
+    # keep_extension is keeping the .df but not .nii; need to figure out
```

Comment:
This looks like it might be a bug in `name_template` parsing. (So possibly worth considering along with any fixes for #2506.)